src/components/contribute/index.js

```diff
@@ -431,11 +432,19 @@ export class Contribute extends React.Component {
     const { DECIMAL_PLACES } = weiToSend.constructor.config()
     weiToSend.constructor.config({ DECIMAL_PLACES: +tokenStore.decimals })
 
-    const tokensToContribute = weiToSend.div(crowdsalePageStore.rate).toFixed()
     weiToSend.constructor.config({ DECIMAL_PLACES })
 
+    const userBalanceBeforeBuy = await getUserBalance(addr, execID, account)
```

Comment:
@mariano-aguero but we already know, what is the balance of a user before contribution. We display it on the page. Thus, we can get it somewhere from the store. There is no need to get it once more.

Reply:
@vbaranov Great.Thanks for the feedback. I added a function that returns the value of the store before purchase and also solve conflicts